### DISCUSSION TITLE
Why are rows and column references transposed in mojo matrix implementations?

### DISCUSSION THREAD

**Discussion by cheburakshu** [modular/modular#858] [Q&A]
Source: https://docs.modular.com/mojo/notebooks/Matmul.html#adding-types-to-the-python-implementation 

```
fn matmul_naive(C: Matrix, A: Matrix, B: Matrix):
    for m in range(C.rows):
        for k in range(A.cols):
            for n in range(C.cols):
                C[m, n] += A[m, k] * B[k, n]
```

`A[m, k]` in the above code appears to reference index `m` for `row` and `k` for `column`, whereas in the `__getitem__` method, it is transposed. There are other examples in documentation where this is happening too, but is there any specific reason the implementation is transposing row and column references?

```
    @always_inline
    fn __getitem__(self, y: Int, x: Int) -> Float32:
        return self.load[1](y, x)
```

**Comment by gdonval** [comment]
That's a typical case of bad naming: it's not transposed. `y` is the row index. If you have a matrix like that:

    [[a, b, c],
     [d, e, f]]

there are 2 rows and 3 columns. `d` is on `[1, 0]`, i.e. `y=1` and `x=0` so `1 * 3 + 0 = 3`, which is the correct index (in 0-based indexing).

Getitem is defined with `y, x`, not the other way around.
